source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -13,19 +13,103 @@
 from comtypes import COMError
 from UIAUtils import isTextRangeOffscreen
 from . import UIATextInfo
-from ..behaviors import KeyboardHandlerBasedTypedCharSupport
+from ..behaviors import EnhancedTermTypedCharSupport, KeyboardHandlerBasedTypedCharSupport
 from ..window import Window
 
 
 class consoleUIATextInfo(UIATextInfo):
-
 	def __init__(self, obj, position, _rangeObj=None):
+		collapseToEnd = None
 		# We want to limit  textInfos to just the visible part of the console.
 		# Therefore we specifically handle POSITION_FIRST, POSITION_LAST and POSITION_ALL.
+		if not _rangeObj and position in (
+			textInfos.POSITION_FIRST,
+			textInfos.POSITION_LAST,
+			textInfos.POSITION_ALL
+		):
+			_rangeObj, collapseToEnd = self._getBoundingRange(obj, position)
+		super(consoleUIATextInfo, self).__init__(obj, position, _rangeObj)
+		if collapseToEnd is not None:
+			self.collapse(end=collapseToEnd)
+
+	def _getBoundingRange(self, obj, position):
+		"""Returns the UIA text range to which the console should be bounded,
+		and whether the textInfo should be collapsed after instantiation."""
+		# microsoft/terminal#4495: In newer consoles,
+		# IUIAutomationTextRange::getVisibleRanges returns a reliable contiguous range.
+		_rangeObj = obj.UIATextPattern.GetVisibleRanges().GetElement(0)
```

Comment:
Could you please put this in a try/except block and raise a RuntimeError when this fails? It should not fail, but better safe than sorry. COMErrors raised directly or indirectly from TextInfo.__init__ usually aren't caught by callers of makeTextInfo.